.gitignore

```diff
@@ -1,5 +1,42 @@
-.idea
```

Comment:
Overall I don't think the .gitignore changes are needed, the current file is fine as is. The rest of the PR looks ok to me.